runtime/lua/vim/lsp/util.lua

```diff
@@ -774,6 +789,10 @@ function M.rename(old_fname, new_fname, opts)
     api.nvim_win_set_buf(win, newbuf)
   end
   api.nvim_buf_delete(oldbuf, { force = true })
+
+  for i, v in ipairs(old_bufs) do
+    api.nvim_buf_delete(tonumber(v), { force = true})
```

Comment:
why force=true? if there are unsaved changes that is not safe. also shouldn't be necessry because of "w!" on line 777

Reply:
I think "w!" on line 777 won't work in this case, because of `if vim.fn.isdirectory(old_fname) == 1`. I have changed it to save old buffers as well